crates/uv/src/commands/project/lock.rs

```diff
@@ -691,17 +753,19 @@ impl ValidatedLock {
             };
         }
 
-        match upgrade {
-            Upgrade::None => {}
-            Upgrade::All => {
-                // If the user specified `--upgrade`, then we can't use the existing lockfile.
-                debug!("Ignoring existing lockfile due to `--upgrade`");
-                return Ok(Self::Unusable(lock));
-            }
-            Upgrade::Packages(_) => {
-                // If the user specified `--upgrade-package`, then at best we can prefer some of
-                // the existing versions.
-                return Ok(Self::Preferable(lock));
+        if !dry_run {
```

Comment:
If we don't do this check, when we use `--dry-run` with `-U` the caller (`do_lock`) will indicate there was an update even if there wasn't one.

I'm not the happiest with this change as it pollutes the API a tad. That being said, I'm not sure of a better approach (doesn't make sense to include dry_run as a variant/member of one of the enums/structs passed in imo, they all seem to be used quite widely) — open to suggestions!